mantidproject/mantid
# Add CMake flag to enable thread sanitizer

Adds `WITH_TSAN` CMake flag to enable thread sanitizer for Linux.

**To test:**

- Ensure you have `libtsan` installed
- Set `WITH_TSAN` to `ON`
- Build
- Start Mantid (or run unit tests) and note the thread sanitizer output in stdout

This does not require release notes because it's a developer only change.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
